src/main/java/org/openjdk/shenandoah/Snapshot.java

```diff
@@ -260,4 +279,147 @@ public double percentageOfOldRegionsInCollectionSet() {
                 old_in_cset, total_in_cset, old, total);
         return total_in_cset == 0 ? 0 : ((double) (old_in_cset)) / total_in_cset;
     }
+
+    public void stateCounter() {
```

Comment:
Can we make this `private` and have the `Snapshot` constructor call this? Is there a use case for a `Snapshot` without these properties computed?

Reply:
Let me try if I can make it private and let the Snapshot constructor to call it, and I don't think there is a case that `Snapshot` without these properties computed.